### PR TITLE
[#139] Importable nodes

### PR DESCRIPTION
**Pull request by vincentcerri**
The purpose of this issue is to make it such that a text / data file can be read and used in conjunction with issue #138. What this means is that the points will all be placed within a given data file and the dev will have to be able to read the given file format and insert all the points within the graph.

**Acceptance Tests:**
- [x] A data file can be opened
- [x] The contents of the data file are turned into Points
- [x] The points are inserted into the Graph
- [x] The points can be manipulated from the Graph

The importable nodes branch implemented the following:
- addNodesToGraph() method that allows us to add nodes from a file and create a graph. 
- encryptAllInputFiles() method that allows the dev to access and encrypt a data file by file name. 
- createGraph() method that reads a node data file and creates the corresponding graph

### CODE REVIEW

File: app/src/main/java/com/example/scoutconcordia/DES.java [hunk 1]
```diff
@@ -124,19 +124,21 @@ private static String bytesToHex(byte[] bytes) {
         return data;
     }
 
+
```
Comment (Averynder) [inline app/src/main/java/com/example/scoutconcordia/DES.java]: within DES the encryptFile and lockFile appear to have the same functionality with the only difference being the output however I believe it is safe to assume for any encryption / decryption the file should rename named the same. The same applies for decryptFile and unlockFile.

File: app/src/main/java/com/example/scoutconcordia/DES.java [hunk 2]
```diff
@@ -172,15 +174,18 @@ public void unlockFile(String filename)
         
         try
         {
+            // THIS IS NOT WORKING
```
Comment (Averynder) [inline app/src/main/java/com/example/scoutconcordia/DES.java]: Definitely shouldn't merge a PR with this comment

File: app/src/main/java/com/example/scoutconcordia/MapsActivity.java [hunk 3]
```diff
@@ -282,6 +269,64 @@ else if (reason == GoogleMap.OnCameraMoveStartedListener.REASON_API_ANIMATION) {
         }
     }
 
+    // This is the method that will be used to encrypt all of the input files during the app startup.
+    public void encryptAllInputFiles()
```
Comment (Averynder) [inline app/src/main/java/com/example/scoutconcordia/MapsActivity.java]: Doesn't really encrypt all files, should obtain a list of files to encrypt / decrypt (the decrypt functionality should be included) and then work from there

File: app/src/main/java/com/example/scoutconcordia/MapsActivity.java [hunk 4]
```diff
@@ -282,6 +269,64 @@ else if (reason == GoogleMap.OnCameraMoveStartedListener.REASON_API_ANIMATION) {
...
+            fos.close();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // this method will be used for creating the floor graphs by reading form a node encrypted text file.
+    public void createGraph(Graph graphName, String encryptedFileName)
```
Comment (Averynder) [inline app/src/main/java/com/example/scoutconcordia/MapsActivity.java]: Having this method is great but it should be refactored to the graph class in the case that we need to call this method from another class / activity. Everything pertaining to the graph should be in the graph class / atomic

File: app/src/main/java/com/example/scoutconcordia/DES.java (hunk 2 above)
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/DES.java]: outdated.

File: app/src/main/java/com/example/scoutconcordia/MapsActivity.java (hunk 3 above)
Comment (vincentcerri) [reply app/src/main/java/com/example/scoutconcordia/MapsActivity.java]: Fixed this issue by creating a file called "filestoencrypt" in the raw folder. This file contains a list of filenames of files that we would like to encrypt. We can add more file names to this as we progress further. The data files should be place within the res/raw folder.

**Pull request merged**